examples/platforms/nrf528xx/src/radio.c

```diff
@@ -452,19 +452,35 @@ otError otPlatRadioReceive(otInstance *aInstance, uint8_t aChannel)
 
 otError otPlatRadioTransmit(otInstance *aInstance, otRadioFrame *aFrame)
 {
-    bool result = true;
+    bool    result = true;
+    otError error  = OT_ERROR_NONE;
 
     aFrame->mPsdu[-1] = aFrame->mLength;
 
     nrf_802154_channel_set(aFrame->mChannel);
 
-    if (aFrame->mInfo.mTxInfo.mCsmaCaEnabled)
+#if OPENTHREAD_CONFIG_THREAD_VERSION >= OT_THREAD_VERSION_1_2
+    if ((otPlatRadioGetCaps(NULL) & OT_RADIO_CAPS_TRANSMIT_TIMING) != 0 && aFrame->mInfo.mTxInfo.mTxDelay != 0)
```

Comment:
```suggestion
    if (aFrame->mInfo.mTxInfo.mTxDelay != 0)
```
I think there are no need to check radio cap in radio layer.